magento/magento2
# Allow to get currect redirect url of Result/Redirect

### Description

During my development I created some plugins to change the redirects of the Controllers:

- `Magento\Customer\Controller\Account\EditPost`
- `Magento\Customer\Controller\Address\FormPost`
- `Magento\Newsletter\Controller\Manage`

The redirects should be changed only in some cases. In an Around- or After-Plugin it was impossible to get the defined redirect url because `\Magento\Framework\Controller\Result\Redirect::$url` is protected and there is no public getter.

### Manual testing scenarios (*)

1. Create a custom module `MyVendor\MyModule`
2. Add di.xml
3. Write a plugin preference f.e.
```
    <type name="Magento\Customer\Controller\Account\EditPost">
        <plugin
            name="CustomEditPost"
            type="MyVendor\MyModule\Plugin\Controller\Customer\Account\EditPostPlugin" />
    </type>
```
4. In the Plugin `MyVendor\MyModule\Plugin\Controller\Customer\Account\EditPostPlugin` it should be possible now to do:
```
    public function aroundExecute(EditPost $subject, \Closure $proceed)
    {
        $resultRedirect = $proceed();
        $url = $resultRedirect->getUrl();
```


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#28954: Allow to get currect redirect url of Result/Redirect
